lib/IRGen/GenStruct.h

```diff
@@ -34,7 +35,19 @@ namespace irgen {
   class IRGenFunction;
   class IRGenModule;
   class MemberAccessStrategy;
-  
+
+  /// If this type has a CXXDestructorDecl, find it and return it. Otherwise,
+  /// return nullptr.
+  clang::CXXDestructorDecl *getDestructor(SILType type);
+
+  /// Find the function address of the given CXXDestructorDecl in clang.
+  llvm::Constant *getDestructorCallee(IRGenModule &IGM,
+                                      clang::CXXDestructorDecl *destructor);
+
+  /// Try to emit a call of the destructor of \p address. If there's no
+  /// destructor, return false. Otherwise, create the call and return true.
+  bool callDestructor(IRGenFunction &IGF, Address address, SILType type);
```

Comment:
This function seems to be only used in one cpp file, can we make it a static function there?

Reply:
Done. I made it a static member of `ClangRecordTypeInfo` (the place it's used).